HW1/P5/P5.py

```diff
@@ -0,0 +1,54 @@
+from pyspark import SparkContext
+from P5_bfs import ssBFS2
+
+# Setup files and context
+sc = SparkContext("local", "P5")
+links = sc.textFile ('s3://Harvard-CS205/wikipedia/links-simple-sorted.txt')
+page_names = sc.textFile ('s3://Harvard-CS205/wikipedia/titles-sorted.txt')
+
+# Set up logger so as to be able to see output
+logger = sc._jvm.org.apache.log4j
+logger.LogManager.getLogger("org").setLevel( logger.Level.ERROR )
+logger.LogManager.getLogger("akka").setLevel( logger.Level.ERROR )
+
+# Functor to create a component for the graph by parsing provided link string
+def createGraphComponent(link):
+	frm, to = link.split(": ")
+	return (int(frm), map(int, to.split(" ")))
+
+# Use functor to create a graph of links similar to P4 graph
+graph = links.map(lambda link: createGraphComponent(link)).cache()
+print graph.collect()
+
+# Zip up page names and take into account 1-indexing
+page_names = page_names.zipWithIndex().map(lambda (page, i): (page, i+1)).sortByKey()
+# Use this one for index lookups
+page_names_idx = page_names.zipWithIndex().map(lambda (page, i): (i+1, page)).sortByKey()
```

Comment:
Great try by using sortBykey before lookup! This could significantly improve lookup efficiency.